packages/adp-tooling/templates/project/package.json

```diff
@@ -15,14 +15,15 @@
         "@sap-ux/ui5-proxy-middleware": "^1.3.0",
         "@sap-ux/deploy-tooling": "^0.11.7"<%}%>,
         "@ui5/task-adaptation": "^1.3.0",
-        "@ui5/cli": "^3.9.2"
+        "@ui5/cli": "^3.9.2"<%if (locals.deploy) {%>,
+        "rimraf": "^5.0.5"<%}%>
     },
     "scripts": {
         "build": "ui5 build --exclude-task generateFlexChangesBundle generateComponentPreload minify --clean-dest",
         "start": "<%= locals.options?.fioriTools ? 'fiori run' : 'ui5 serve' %> --open /test/flp.html#app-preview",
         "start-editor": "<%= locals.options?.fioriTools ? 'fiori run' : 'ui5 serve' %> --open /test/adaptation-editor.html"<%if (locals.deploy) {%>,
-        "deploy": "ui5 build --config ui5-deploy.yaml --exclude-task <%= locals.options?.fioriTools ? 'deploy-to-abap' : 'abap-deploy-task' %> generateFlexChangesBundle generateComponentPreload --clean-dest && <%= locals.options?.fioriTools ? 'fiori deploy' : 'deploy' %> --config ui5-deploy.yaml",
-        "undeploy": "<%= locals.options?.fioriTools ? 'fiori undeploy' : 'undeploy' %> --config ui5-deploy.yaml --lrep \"apps/<%= app.reference %>/appVariants/<%= app.id %>/\"",
+        "deploy": "npm run build && fiori deploy --config ui5-deploy.yaml && rimraf archive.zip",
```

Comment:
The deploy process should not generate an `archive.zip` if not explicitly requested, therefore, I am confused that you require `rimraf` here.
If it does generate an `archive.zip` then there is a regression that needs an investigation.

Reply:
You also need to add the `IF` again because you only have the `fiori deploy` command if the `Fiori tools` are added, otherwise, you need to use `ui5 build`

Reply:
@tobiasqueck in SAP Fiori tools, the package.json deploy script is appended as follows;
```
"deploy": "npm run build && fiori deploy --config ui5-deploy.yaml && rimraf archive.zip",
```
During the deploy task, there is a zip generated on the fly. However, a user can specify a path to an already generated zip which will obviously bypass this zip generation task.

https://github.com/SAP/open-ux-tools/blob/main/packages/deploy-tooling/src/cli/archive.ts#L82

Reply:
I checked in @sap-ux/abap-deploy-config-writer and this module is always  generating the same scripts  like in this PR

Reply:
Well ... I still don't understand the usage of `rimraf` because as @longieirl said, the zip is generated on the fly, so nothing needs to be deleted.
The deploy config generated is only generating things for the Fiori tools, while the adp writer already supports both, Fiori tools and a "pure" open source version, so instead of introducing a regression in the adp-writer, this needs to be improved in the deploy-config writer, and then, maybe reused here instead of copying an implementation

Reply:
@tobiasqueck the archive remains if there is any exception thrown during the process! I think this is more of a cleanup task.

Reply:
Fixed in [a3c9caa](https://github.com/SAP/open-ux-tools/pull/2362/commits/a3c9caada35cc2c1ce209f4e86e7b03a04f73ba5)

Reply:
Not sure I understand what was fixed with https://github.com/SAP/open-ux-tools/pull/2362/commits/a3c9caada35cc2c1ce209f4e86e7b03a04f73ba5. I am expecting that we add `--clean-dest` again and remove `rimraf`. I think it is a good idea to keep the zip if something went wrong with the deployment, if all goes well, no zipfile is created if I understood @longieirl correctly.